lib/FSi/DoctrineExtensions/Translatable/TranslationHelper.php

```diff
@@ -182,43 +176,27 @@ public function getObjectLocale(ClassTranslationContext $context, $object)
      */
     private function setObjectLocale(TranslatableClassMetadata $classMetadata, $object, $locale)
     {
-        $localeProperty = $classMetadata->localeProperty;
-        $this->getPropertyAccessor()->setValue($object, $localeProperty, $locale);
+        $this->propertyManipulator->setPropertyValue($object, $classMetadata->localeProperty, $locale);
     }
 
     /**
      * @param object $source
      * @param object $target
      * @param array $properties
      */
-    private function copyProperties($source, $target, $properties, ClassMetadata $classMetadata)
+    private function copyProperties($source, $target, $properties)
     {
-        $propertyAccessor = $this->getPropertyAccessor();
         foreach ($properties as $sourceField => $targetField) {
-            if ($classMetadata->isCollectionValuedAssociation($targetField)
-                && $propertyAccessor->getValue($target, $targetField) === null
-            ) {
-                $reflection = new \ReflectionClass($target);
-                $property = $reflection->getProperty($targetField);
-                $property->setAccessible(true);
-                $property->setValue($target, new ArrayCollection());
-                $property->setAccessible(false);
+            $sourceProperty = $this->propertyManipulator->getPropertyValue($source, $sourceField);
+            if ($sourceProperty instanceof PersistentCollection) {
+                $sourceProperty->initialize();
```

Comment:
There should be a bunch of tests that verify if collection created/modified throught translatable/translation object is readable from translation/translatable object after flushing and clearing entity manager. These tests will show if this initialization is necessary.